src/VisualStudio/IntegrationTest/TestUtilities/InProcess/VisualStudioWorkspace_InProc.cs

```diff
@@ -109,10 +110,31 @@ private static TestingOnly_WaitingService GetWaitingService()
             => GetComponentModel().DefaultExportProvider.GetExport<TestingOnly_WaitingService>().Value;
 
         public void WaitForAsyncOperations(TimeSpan timeout, string featuresToWaitFor, bool waitForWorkspaceFirst = true)
-            => GetWaitingService().WaitForAsyncOperations(timeout, featuresToWaitFor, waitForWorkspaceFirst);
+        {
+            if (waitForWorkspaceFirst || featuresToWaitFor == FeatureAttribute.Workspace)
+            {
+                WaitForProjectSystem(timeout);
```

Comment:
@sharwell Is there a reason this code isn't in the implementation of GetWaitingService? Because don't we have other clients (DDRITs?) calling into that?

Reply:
The waiting service doesn't have visibility to the layers where the operation progress APIs are.